app/src/main/java/org/p2p/wallet/infrastructure/network/alarmlogger/AlarmErrorsLogger.kt

```diff
@@ -0,0 +1,102 @@
+package org.p2p.wallet.infrastructure.network.alarmlogger
+
+import com.google.gson.Gson
+import timber.log.Timber
+import kotlinx.coroutines.launch
+import org.p2p.wallet.BuildConfig
+import org.p2p.wallet.common.di.AppScope
+import org.p2p.wallet.infrastructure.network.alarmlogger.api.AlarmErrorsRequest
+import org.p2p.wallet.infrastructure.network.alarmlogger.api.AlarmErrorsServiceApi
+import org.p2p.wallet.infrastructure.network.alarmlogger.api.AlarmErrorsSwapRequest
+import org.p2p.wallet.infrastructure.network.data.ServerException
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.jupiter.repository.transaction.JupiterSwapTransactionMapper
+import org.p2p.wallet.jupiter.statemanager.SwapState
+import org.p2p.wallet.utils.Base58String
+import org.p2p.wallet.utils.retryRequest
+import org.p2p.wallet.utils.toBase58Instance
+import org.p2p.wallet.utils.toJsonObject
+
+class AlarmErrorsLogger(
+    private val api: AlarmErrorsServiceApi,
+    private val tokenKeyProvider: TokenKeyProvider,
+    private val transactionMapper: JupiterSwapTransactionMapper,
+    private val gson: Gson,
+    private val appScope: AppScope
+) {
+    private class AlarmErrorsError(override val cause: Throwable) : Throwable(cause)
+
+    enum class SwapAlarmError {
+        BLOCKCHAIN_ERROR, LOW_SLIPPAGE, UNKNOWN
+    }
+
+    private val userPublicKey: Base58String
+        get() = tokenKeyProvider.publicKey.toBase58Instance()
+
+    private val isReleaseBuild: Boolean
+        get() = BuildConfig.BUILD_TYPE == "release"
+
+    fun sendSwapAlarm(
+        swapErrorType: SwapAlarmError,
+        swapState: SwapState.SwapLoaded,
+        swapError: Throwable
+    ) {
+        if (isReleaseBuild) {
+            appScope.launch {
+                try {
+                    val errorTitle = getSwapErrorTitle(swapErrorType)
+                    val request = gson.toJsonObject(createSwapRequest(swapState, swapError))
+                        .let {
```

Comment:
you can drop down `.let { }` 
`val requestJson = gson.toJsonObject(createSwapRequest(swapState, swapError))`
`val request  = AlarmErrorsRequest(
                                logsTitle = "Swap Android Alarm ($errorTitle)",
                                payload = requestJson.toString()
                            )`